hs_core/admin.py

```diff
@@ -1,11 +1,54 @@
 from django import forms
-from django.contrib.auth.admin import UserAdmin
+from django.contrib import messages
+from django.contrib.auth.admin import UserAdmin as DjangoUserAdmin
+from django.contrib.admin.actions import delete_selected as django_delete_selected
 from django.contrib.auth.forms import UserCreationForm
 from django.contrib.gis import admin
 from django.contrib.contenttypes.admin import GenericTabularInline
 from django.utils.translation import gettext_lazy as _
 
-from .models import ResourceFile, User
+from .models import ResourceFile, User, BaseResource, Creator
+
+
+class UserAdmin(DjangoUserAdmin):
+    actions = ['delete_selected']
+
+    def delete_model(self, request, obj):
+        # prevent user delete if user is an owner/author on a published resource
+        try:
+            c = Creator.objects.get(hydroshare_user_id=obj.id)
+            published_resources = BaseResource.objects.filter(object_id=c.content_object.id, raccess__published=True)
+        except Creator.DoesNotExist:
+            published_resources = None
+        if published_resources:
+            res_ids = ", ".join(str(res.short_id) for res in published_resources)
+            message = f"Can't delete user. They are a creator of published resource(s): {res_ids}"
+            self.message_user(request, message, messages.ERROR)
+            # https://github.com/django/django/blob/3.2/django/contrib/admin/actions.py#L46-L48C27
+            return False
```

Comment:
I couldn't find a way to prevent the default success message:
![image](https://github.com/hydroshare/hydroshare/assets/17934193/51f62a34-d6ca-4fff-bd1f-6b664f0e82a7)